File: haskell/Problem15.hs
-- Copyright 2015 Mitchell Kember. Subject to the MIT License.
-- Project Euler: Problem 15
-- Lattice paths

module Problem15 where

import Common (combinations)

routes :: Int -> Int -> Int
routes w h = fromIntegral result
  where
    ww = toInteger w
    hh = toInteger h
    result = combinations ww (ww + hh)

solve :: Int
solve = routes 20 20

File: c++/src/problem_15.cpp
// Copyright 2016 Mitchell Kember. Subject to the MIT License.
// Project Euler: Problem 15
// Lattice paths

#include "common.hpp"

namespace problem_15 {

long lattice_paths(const long w, const long h) {
	return common::combinations(w + h, w);
}

long solve() {
	return lattice_paths(20, 20);
}

} // namespace problem_15
